irods/session.py

```diff
@@ -113,7 +115,9 @@ def _configure_account(self, **kwargs):
         except KeyError:
             pass
 
-        creds['password'] = self.get_irods_password(session_ = self, **creds)
+        pw = creds['password'] = self.get_irods_password(session_ = self, **creds)
+        if not pw and creds.get('irods_user_name') != 'anonymous':
+            raise NonAnonymousLoginWithoutPassword
```

Comment:
Are we now requiring that `.irodsA` (or the authentication file) exist in order to use PRC? Or has this always been the case and now we are making it more clear?

Also, maybe consider adding a message to the exception?

Reply:
Used to be (before 91b1adaa7c2e78c867407cd55008fe4c75de0ed5) that a FileNotFoundError was raised if no .irodsA were encountered in the expected place. (As dictated by ~/.irods or the IRODS_AUTHENTICATION_FILE environment variable.)  Now we're raising this exception, as it's - I think - less confusing and delineates the difference between the two cases (anonymous, which needs no .irodsA, and other logins, which do).

Reply:
But we could include an exception message of 'expected file at <.irodsA-path>' ... would that be appropriate ?

Reply:
that seems nice.    and we'd know, even if it was a non-standard or environment-variable-provided location?

Reply:
agreed